pkg/controller/v1beta1/inferenceservice/reconcilers/knative/ksvc_reconciler.go

```diff
@@ -227,6 +225,19 @@ func (r *KsvcReconciler) Reconcile() (*knservingv1.ServiceStatus, error) {
 		}
 		return nil, err
 	}
+	// Set ResourceVersion which is required for update operation.
+	desired.ResourceVersion = existing.ResourceVersion
```

Comment:
@yuzisun From K8s docs: You can overwrite (update) an existing resource - for example, a ConfigMap - using an HTTP PUT. For a PUT request, it is the client's responsibility to specify the resourceVersion (taking this from the object being updated). Kubernetes uses that resourceVersion information so that the API server can detect lost updates and reject requests made by a client that is out of date with the cluster. In the event that the resource has changed (the resourceVersion the client provided is stale), the API server returns a 409 Conflict error response.

https://kubernetes.io/docs/reference/using-api/api-concepts/#patch-and-apply

https://github.com/argoproj/argo-cd/issues/3657#issuecomment-722706739

https://github.com/kubernetes-sigs/hierarchical-namespaces/issues/53#issuecomment-874186845

https://github.com/kubernetes-sigs/hierarchical-namespaces/issues/53#issuecomment-874751106